### PR TITLE
Seperate drawMap from zoomToPosition and add configuration for clusters

### PR DESCRIPTION
**Pull request by alexclarkofficial**
In reference to: emberjs#1904

### CODE REVIEW

**Comment by wifelette** [comment]
Thumbs up on this from @wycats (in theory, we just looked at the code, haven't pulled it and tested :stuck_out_tongue:) / cc @sandraor

**Comment by alexclarkofficial** [comment]
Do we have the PNG's we would like to use for the clusters? This is the most basic implementation, but I'd be happy to do the CSS to make it look right.

**Comment by alexclarkofficial** [comment]
Oh, also I just figured out how to set the break points for cluster size. If we'd like to change cluster images at different sizes let me know (I think it's currently set to 2 and 10).

**Comment by wifelette** [comment]
RE: image for clusters: best answered by @sandraor

**Comment by sandraor** [comment]
Thanks, @alexclarkofficial! Here are the two cluster images. The first one is designed for 1 digit, the second one for 2 digits, both with white text. The smallest breakpoint (2) seems too aggressive, so can we try it at 4?

map-cluster-1_x2
map-cluster-1
map-cluster-2_x2
map-cluster-2

**Comment by alexclarkofficial** [comment]
Thanks @sandraor! I went with the smaller image for each cluster size that you provided. I tried smallest breakpoints of both 3 and 4. You can check out the results below. Let me know what you think and any other remarks about styling and I'll make the commit.

Globe with smallest breakpoint of 3:
http://cl.ly/image/1o3T061Y0z0w

Globe with smallest breakpoint of 4:
http://cl.ly/image/3s2e2I0n2s1Y

Europe with smallest breakpoint of 3:
http://cl.ly/image/2X2a0a161B1p

Europe with smallest breakpoint of 4:
http://cl.ly/image/2L242p0h2m0L

**Comment by wifelette** [comment]
I'm way too excited to get this merged in :grinning:

File: source/javascripts/meetups.js [hunk 1]
```diff
@@ -1,5 +1,25 @@
 (function() {
-  var handler = Gmaps.build('Google'),
+  var handler = Gmaps.build('Google', {
+        markers: {
```
Comment (wycats) [inline source/javascripts/meetups.js]: Something seems weird about the indentation style here.

File: source/javascripts/meetups.js [hunk 2]
```diff
@@ -44,37 +64,35 @@
   locations.forEach( generateMarkerData );
 
   handler.buildMap(mapOptions, function() {
```
Comment (wycats) [inline source/javascripts/meetups.js:67]: I think I saw this updated by @solsberg. Is that right?

File: source/javascripts/meetups.js [hunk 3]
```diff
@@ -44,37 +64,35 @@
...
+        zoomToPosition(position);
+      });
     }
-  } );
+  });
 
   function bindLiToMarker(json_array) {
     _.each(json_array, function(json){
 
-        var markerId = json.location.toLowerCase().replace(/\W/g, '');
```
Comment (wycats) [inline source/javascripts/meetups.js:57]: This is just an indentation change, right?

Comment (alexclarkofficial) [inline source/javascripts/meetups.js:57]: Yep.

File: source/javascripts/meetups.js (hunk 1 above)
Comment (alexclarkofficial) [inline source/javascripts/meetups.js]: That is weird - it will be fixed in my next commit.

File: source/javascripts/meetups.js (hunk 2 above)
Comment (alexclarkofficial) [inline source/javascripts/meetups.js:67]: It looks like he did something along the same lines, which I had missed. How should I proceed?

Comment (wifelette) [inline source/javascripts/meetups.js:67]: @wycats @sandraor Thoughts on how to resolve? Presumably we need to pick one and adapt the other PR to accommodate.

Comment (wifelette) [inline source/javascripts/meetups.js:67]: Closing the loop here: @alexclarkofficial and @solberg talked on the other PR (https://github.com/sandraor/website/pull/10/files#r22864632) and @solberg is going to remove the conflicting code from his PR :)

**Comment by wifelette** [comment]
Making sure we're all on the same page here: @alexclarkofficial is fixing a couple indentation changes, and then this is good to be merged. He'll comment here (and/or here too: https://github.com/emberjs/website/pull/1904) to let @sandraor know when https://github.com/sandraor/website/pull/9 is ready to be merged in.

**Comment by alexclarkofficial** [comment]
Thanks @wifelette! I was hoping to get @sandraor's design approval and final comment on the smallest breakpoint for clusters - but other than that I can make the commit anytime.

**Comment by sandraor** [comment]
I think the breakpoint at 4 looks best. Thanks for testing them out!

Also, I just noticed that github didn't keep or display the names of the cluster images I uploaded, so I'm sorry, what I said about them didn't make much sense! :( I think only the larger image needs to be changed:

This is for the small breakpoint (map-cluster-1.png): https://cloud.githubusercontent.com/assets/2864198/5715531/599d9ba8-9aa8-11e4-9537-45c10012ac8c.png
This is the retina version (map-cluster-1_x2.png):
https://cloud.githubusercontent.com/assets/2864198/5715530/598a88a6-9aa8-11e4-902a-476fcf2e8bf9.png

This is for the large breakpoint (map-cluster-2.png): https://cloud.githubusercontent.com/assets/2864198/5715533/59a443c2-9aa8-11e4-92fc-edd75c434146.png
This is the retina version (map-cluster-2_x2.png): https://cloud.githubusercontent.com/assets/2864198/5715532/59a2efa4-9aa8-11e4-8091-d15466a6d62f.png

Then I think it's ready to go! :D

**Comment by alexclarkofficial** [comment]
Changes have been made, but unfortunately I couldn't figure out how to include the URLs for the retina images. The doc's showed a boolean for enabling retina images, but no details on how to include them.
http://google-maps-utility-library-v3.googlecode.com/svn/trunk/markerclustererplus/docs/reference.html#ClusterIconInfo

**Comment by sandraor** [comment]
No worries — I just included them in case they were needed. 

This looks great! Thank you! :)

**Pull request merged**